File: arrays/plus_one/plus_one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        convertedLong = long(''.join(map(str,digits)))
        plusOne = convertedLong + 1
        
        convertedString = str(plusOne)
        result = [None]*len(convertedString)
        for index, value in enumerate(convertedString):
            result[index] = int(value)
        
        return result

File: arrays/plus_one/plus_one.go
package main

func plusOne(digits []int) []int {
    length := len(digits)

    for length != 0 {
        length--
        digits[length] += 1
        if digits[length] > 9 {
            digits[length] = 0
        } else {
            return digits
        }
    }

    newLength := len(digits)+1
    result := make([]int, newLength)
    result[0] = 1
    for i := 1; i<newLength; i++ {
        result[i] = digits[i-1]
    }

    return result
}
